sponge/src/main/java/net/buycraft/plugin/sponge/BuycraftPlugin.java

```diff
@@ -0,0 +1,194 @@
+                updateInformation(client);
+            } catch (IOException | ApiException e) {
+                getLogger().error(String.format("We can't check if your server can connect to Buycraft: %s", e.getMessage()));
+            }
+            apiClient = client;
+        }
+        placeholderManager.addPlaceholder(new NamePlaceholder());
+        placeholderManager.addPlaceholder(new UuidPlaceholder());
+        platform.executeAsyncLater(duePlayerFetcher = new DuePlayerFetcher(platform), 1, TimeUnit.SECONDS);
+        commandExecutor = new QueuedCommandExecutor(platform);
+        Sponge.getScheduler().createTaskBuilder().intervalTicks(1).delayTicks(1).execute((Runnable) commandExecutor).async().submit(this);
+        listingUpdateTask = new ListingUpdateTask(this);
+        if (apiClient != null) {
+            getLogger().info("Fetching all server packages...");
+            listingUpdateTask.run();
+            Sponge.getScheduler().createTaskBuilder().delayTicks(20 * 60 * 20).intervalTicks(20 * 60 * 20).execute(listingUpdateTask).async()
+                    .submit(this);
+        }
+
+        Sponge.getEventManager().registerListeners(this, new BuycraftListener(this));
+
+        /*
+        *TODO: Commands.
+         */
+        Sponge.getCommandManager().register(this, buildCommands(), "buycraft", "buy");
+
+
+    }
+
+    private CommandSpec buildCommands() {
+        CommandSpec refresh =
+                CommandSpec.builder()
+                        .description(Text.of("Refreshes the package listing."))
+                        .permission("buycraft.admin")
+                        .executor(new RefreshCmd(this))
+                        .build();
+        CommandSpec secret =
+                CommandSpec.builder()
+                        .description(Text.of("Sets the secret key to use for this server."))
+                        .permission("buycraft.admin")
+                        .arguments(GenericArguments.string(Text.of("secret")))
+                        .executor(new SecretCmd(this))
+                        .build();
+        CommandSpec report =
+                CommandSpec.builder()
+                        .description(Text.of("Generates a report with debugging information you can send to support."))
+                        .executor(new ReportCmd(this))
+                        .permission("buycraft.admin")
+                        .build();
+        CommandSpec list = CommandSpec.builder()
+                .description(Text.of("Lists all Buycraft packages and their prices."))
+                .executor(new ListPackagesCmd(this))
+                .build();
+        return CommandSpec.builder()
+                .description(Text.of("Main command for the Buycraft plugin."))
+                .child(report, "report")
+                .child(list, "list", "packages")
+                .child(secret, "secret")
+                .child(refresh, "refresh")
+                .executor(new GuiCmd(this))
```

Comment:
Looks like left-over stuff. GUI isn't supported. Even then, the command is defined separately.